core/src/main/scala/org/bitcoins/core/number/NumberType.scala

```diff
@@ -209,10 +209,66 @@ final case class UInt64(underlying: BigInt)
     with BasicArithmetic[UInt64]
     with Ordered[UInt64] {
 
-  override def hex: String = BitcoinSUtil.encodeHex(underlying)
+  override def hex: String = encodeHex(underlying)
 
   override def bytes: ByteVector = ByteVector.fromValidHex(hex)
 
+  override def +(y: UInt64): UInt64 = {
+    UInt64(underlying + y.underlying)
+  }
+
+  override def -(y: UInt64): UInt64 = {
+    UInt64(underlying - y.underlying)
+  }
+
+  override def *(y: UInt64): UInt64 = {
+    UInt64(underlying * y.underlying)
+  }
+
+  override def *(factor: BigInt): UInt64 = ???
+
+  def |(u64: UInt64): UInt64 = {
+    UInt64(underlying | u64.underlying)
+  }
+
+  def &(u64: UInt64): UInt64 = {
+    UInt64(underlying & u64.underlying)
+  }
+
+  def <<(u64: UInt64): UInt64 = {
+    UInt64(underlying << u64.underlying)
+  }
+
+  def <<(int: Int): UInt64 = {
+    this.<<(UInt64(int))
+  }
+
+  def >>(u64: UInt64): UInt64 = {
+    UInt64(underlying >> u64.underlying)
+  }
+
+  def >>(int: Int): UInt64 = {
+    UInt64(underlying >> int)
+  }
+
+  def toInt: Int = {
+    val i = underlying.toInt
+    require(underlying == i,
+            s"Rounded when converting long=${underlying} toInt=${i}")
+    i
+  }
+
+  def toBigInt: BigInt = underlying
+
+  def toLong: Long = toBigInt
+
+  def toDouble(x: UInt64): Double = x.underlying.toDouble
```

Comment:
`def toDouble(x: UInt64)`, `def toFloat(x: UInt64)` and `def toBigInt(x: UInt64)` doesn't really make much sense to have as method on this trait. Why would a `NumberType` be responsible for converting external values?